### ISSUE TITLE
add api to obtain or ignore invalid lines

### ISSUE THREAD

**Issue by wbolster** [wbolster/jsonlines#7]
e.g. for reading json-formatted log files with interspersed non-json lines

**Comment by wbolster** [comment]
the `.read()` method could raise a custom exception with a custom attribute containing the original value.

however, this results in ugly code patterns in typical iterator-style stream consumption, so as an alternative an extra flag to some `.iter(...)` helper might be useful. the items yielded during iteration could be `(success, value)` tuples, so that consuming code does not need any verbose exception handling and manual iterator advancing:

```python
for success, value in reader.iter(allow_invalid=True):
    if success:
        pass  # yay
    else:
        pass # yolo
```

additionally, it could have a `.iter(skip_invalid=True)` flag to simply ignore invalid lines.

**Comment by wbolster** [comment]
exception `.value` attribute partly implemented in f0b643d

**Comment by wbolster** [comment]
fixed as part of e0bc0f81f54d2b7d1d7a45da4ff3fccb66e9a5bb
